logic/common/tiled_utils.py

```diff
@@ -4,72 +4,70 @@
+      tiles2d     - 2d array of Tile IDs, first row is at top
+      >>> DecodeIntoTiles2d("eAENw4cNACAMAKA66/r/XiGhRES12R1O0+X2eH1+BeAATw==", 4)
+      >>> [[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]]
+    '''
+
+    # Convert the base64 string to a numpy array (uint32)
     decoded_data = base64.b64decode(encoded_str)
-    
-    # Decompress the data using zlib
     decompressed_data = zlib.decompress(decoded_data)
-    
-    # Convert the decompressed data to a numpy array with type uint32
     array = numpy.frombuffer(decompressed_data, dtype=numpy.uint32)
-    
+
     # Reshape the array to the desired dimensions
     tiles2d = array.reshape(-1, row_width)
-    
     return tiles2d.tolist()
-    
+
+
+
 def EncodeIntoZlibString64(tiles2d):
-    ''' Takes a tiles2d and converts it into a zlib base64 encoded string.
-    
-    :param tiles2d: A 2d array of tile IDs
-    :returns encoded_string: a zlib base64 string
-        
-    Example: 
-    >>> EncodeIntoZlibString64([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
-    >>> "eAENw4cNACAMAKA66/r/XiGhRES12R1O0+X2eH1+BeAATw=="
     '''
-    # Convert the input array to a numpy array with type uint32
-    np_array = numpy.array(tiles2d, dtype=numpy.uint32)
+    Takes a TILED layer encoded string and converts it into a tiles2d
+      tiles2d     - 2d array of Tile IDs, first row is at top
+      encoded_str - string containing a layer's 'data' in TILED, zlib-encoded and base64-compressed
+      >>> EncodeIntoZlibString64([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
+      >>> "eAENw4cNACAMAKA66/r/XiGhRES12R1O0+X2eH1+BeAATw=="
+    '''
 
-    # Convert the numpy array to bytes
+    # Convert: tiles2d array -> numpy array (uint32) -> bytes -> zlib string -> base64 string
+    np_array = numpy.array(tiles2d, dtype=numpy.uint32)
     array_bytes = np_array.tobytes()
-
-    # Compress the bytes using zlib
     compressed_data = zlib.compress(array_bytes)
-
-    # Encode the compressed data using base64
     encoded_str = base64.b64encode(compressed_data).decode()
 
     return encoded_str
 
 
+
+#--------------------------------------------------#
+'''Tiles2D Utilities'''
+
 def PrintTiles2d(tiles2d, multi_line=False):
     '''Prints a tiles2d (2d array of tile IDs). If multi_line is specified, will space out each row.'''
     if multi_line:
          print('\n'.join(f"[{', '.join(map(str, row))}]" for row in tiles2d))
     else:
         print('[' + ', '.join('[' + ', '.join(str(x) for x in row) + ']' for row in tiles2d) + ']')
 
+
+
+
 def TrimTiles2d(tiles2d):
-    '''Removes empty rows and columns from a Tiles2D.
-    
-    :param tiles2d: A 2d array of tile IDs to trim. A cell is considered empty if it is 0
-        
-    Example: 
-    >>> TrimTiles2d([[0, 0, 0], [0, 1, 2], [0, 3, 4]])
-    >>> [[1, 2], [3, 4]]
+    '''
+    Removes empty rows and columns from a Tiles2D.
+      tiles2d     - 2d array of Tile IDs, cells with value 0 are "empty tile"
```

Comment:
I'm not a fan of this type of spacing - it creates unseemly gaps in the description IMO. Let's just use one space after the variable name